database/influxdb/influxdb_user.py

```diff
@@ -0,0 +1,220 @@
+      password: "{{ influxdb_password }}"
+      user_name: "{{ influxdb_user_name }}"
+      user_pass: "{{ influxdb_user_pass }}"
+      user_admin: yes
+      state: present
+'''
+
+RETURN = '''
+#only defaults
+'''
+
+try:
+    from influxdb import InfluxDBClient
+    from influxdb import exceptions
+    HAS_INFLUXDB = True
+except ImportError:
+    HAS_INFLUXDB = False
+
+
+def influxdb_argument_spec():
+    return dict(
+        hostname=dict(required=True, type='str'),
+        port=dict(default=8086, type='int'),
+        database_name=dict(required=True, type='str', default=False),
+        username=dict(default='root', type='str'),
+        password=dict(default='root', type='str', no_log=True),
+        user_name=dict(required=True, type='str'),
+        user_pass=dict(required=True, type='str', no_log=True),
+        user_admin=dict(required=False, default=False, type='bool')
+    )
+
+
+def connect_to_influxdb(module):
+    hostname = module.params['hostname']
+    port = module.params['port']
+    username = module.params['username']
+    password = module.params['password']
+    database_name = module.params['database_name']
+
+    client = InfluxDBClient(
+        host=hostname,
+        port=port,
+        username=username,
+        password=password,
+        database=database_name
+    )
+    return client
+
+
+def find_user(module, client, user_name):
+    u = None
+
+    try:
+        users = client.get_list_users()
+        for user in users:
+            if user['user'] == user_name:
+                u = user
+                break
+    except requests.exceptions.ConnectionError as e:
+        module.fail_json(msg=str(e))
+    return u
+
+def create_user(module, client, user_name, user_pass, user_admin):
+    if not module.check_mode:
+        try:
+            client.create_user(user_name, user_pass, user_admin)
+        except requests.exceptions.ConnectionError as e:
+            module.fail_json(msg=str(e))
+
+    module.exit_json(changed=True)
+
+
+def drop_user(module, client, user_name):
+    if not module.check_mode:
+        try:
+            client.drop_user(user_name)
+        except exceptions.InfluxDBClientError as e:
+            module.fail_json(msg=e.content)
+
+    module.exit_json(changed=True)
+
+
+def main():
+    argument_spec = influxdb_argument_spec()
+    argument_spec.update(
+        state=dict(default='present', type='str', choices=['present', 'absent'])
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    if not HAS_INFLUXDB:
+        module.fail_json(msg='influxdb python package is required for this module')
+
+    state = module.params['state']
+    user_name = module.params['user_name']
+    user_pass = module.params['user_pass']
+    user_admin = module.params['user_admin']
+
+    try:
+        client = connect_to_influxdb(module)
+        user = find_user(module, client, user_name)
```

Comment:
The current logic only checks for the existence of users, and not their permissions. For example, creating a non-admin user and then attempting to create the same user with admin privileges results in:

```
{
    "changed": false,
    "invocation": {
        "module_args": {
            "database_name": "ansible-********",
            "hostname": "localhost",
            "password": "root",
            "port": 8086,
            "state": "present",
            "user_admin": true,
            "user_name": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "user_pass": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "username": "root"
        }
    }
}
```

Where I would expect the current user permissions to be taken into account, and a [grant_admin_privileges](https://github.com/influxdata/influxdb-python/blob/master/influxdb/client.py#L684) call made if the user is not an admin.